### PR TITLE
portable gravity machine

### PR DESCRIPTION
**Pull request by mc-oofert**
## About The Pull Request

https://github.com/user-attachments/assets/98aa2177-ee0f-48d0-bbd4-83d01e688b54

it makes gravity in a radius
you may order it from cargo
may take power from wire instead
## Why It's Good For The Game

good for like space bases and stuff

## Changelog

:cl:
add: portable gravity unit, bought at cargo
/:cl:

### CODE REVIEW

**Comment by moocowswag** [comment]
High quality af

**Comment by Loafin34** [comment]
is there a zone where there's high gravity?

**Comment by Dawnseer** [comment]
Can we increase the gravity it provides? For fitness, that is.

**Comment by Wallemations** [comment]
Any method we could use to visualize the field? Maybe a subtle outline or some subtle falling particles?

**Comment by Xackii** [comment]
Put it somewhere roundstart please

**Comment by mc-oofert** [comment]
> is there a zone where there's high gravity?

it can produce deadly gravity if emagged but it can normally produce nonlethal high gravity

also it couldve made negative grav but thats broken as fuck

**Comment by CCC23** [comment]
Finally I can make the crunchinator and obliterate anyone who walks nears bones

**Comment by CRITAWAKETS** [comment]
this is really cool but a max gravity range of 12 (with full part upgrades) seems a little low for space base shenanigans to use

**Comment by mc-oofert** [comment]
> this is really cool but a max gravity range of 12 (with full part upgrades) seems a little low for space base shenanigans to use

144 tiles is a lot

**Review by Wallemations — CHANGES_REQUESTED** [review]
> Any method we could use to visualize the field? Maybe a subtle outline or some subtle falling particles?

Should have made this part of a review, oops

File: tgui/packages/tgui/interfaces/portagrav.tsx [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+  percentage: number;
+  on: BooleanLike;
+  range: number;
+  maxrange: number;
+  gravity: number;
+  wiremode: BooleanLike;
+  draw: number;
+};
+
+export const portagrav = (props) => {
```
Comment (jlsnow301) [inline tgui/packages/tgui/interfaces/portagrav.tsx:17]: why is this lower case? react components, and their containing files, are pascal case

**Review by jlsnow301 — CHANGES_REQUESTED** [review]
naming issues

**Comment by Wallemations** [comment]
> image
> 
> 

Not a fan tbh
I think just spawning a couple falling particles on each tile would work tbh
Either that or changing the sprite you used for this to instead be falling glowing pixels
That way they're clearly showing it's gravity through downward motion

Also, how do peacekeeper borgs make those rings around themselves? If it's code-based you could try that
Unless it's just a really big sprite lmao

**Comment by mc-oofert** [comment]
> Not a fan tbh I think just spawning a couple falling particles on each tile would work tbh Either that or changing the sprite you used for this to instead be falling glowing pixels That way they're clearly showing it's gravity through downward motion Hell I'll make the sprite for you if you want me to

my ass sucks at spriting so that could do although im not so sure about the other gravity options like (disabled because broken) -1G or 0G

**Comment by Wallemations** [comment]
> > Not a fan tbh I think just spawning a couple falling particles on each tile would work tbh Either that or changing the sprite you used for this to instead be falling glowing pixels That way they're clearly showing it's gravity through downward motion Hell I'll make the sprite for you if you want me to
> 
> my ass sucks at spriting so that could do although im not so sure about the other gravity options like (disabled because broken) -1G or 0G

I don't see those sprites in your icon diffs, so idk where exactly you're gonna put 'em either

I'll figure out the other options, shouldn't be too hard tbh

**Comment by Wallemations** [comment]
Alright I made some border sprites and am working on some particles, what's your discord so I can shuffle those files to you?

**Comment by mc-oofert** [comment]
> Alright I made some border sprites and am working on some particles, what's your discord so I can shuffle those files to you?

actuallyjimmyl, youll have to send a friend request because im permad

**Comment by mc-oofert** [comment]
https://github.com/user-attachments/assets/e54707c8-4eaa-42b4-8980-df43527a7f4c

**Review by Wallemations — APPROVED** [review]

**Review by jlsnow301 — APPROVED** [review]
Ui is fine

File: code/game/machinery/portagrav.dm [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+		return ITEM_INTERACT_SUCCESS
+
+/obj/machinery/power/portagrav/crowbar_act(mob/living/user, obj/item/tool)
+	. = NONE
+	if(default_deconstruction_crowbar(tool))
+		return ITEM_INTERACT_SUCCESS
+
+/obj/machinery/power/portagrav/item_interaction(mob/living/user, obj/item/tool, list/modifiers)
+	. = NONE
+	if(istype(tool, /obj/item/stock_parts/power_store/cell))
```
Comment (Ghommie) [inline code/game/machinery/portagrav.dm:101]: early return

**Review by Ghommie — CHANGES_REQUESTED** [review]

**Comment by Ghommie** [comment]
Also, how does it work with the wallening?

**Comment by mc-oofert** [comment]
> Also, how does it work with the wallening?

it probably works

**Pull request merged**